### PR TITLE
ucloud 0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by lixiaojun629**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/ucloud.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Ucloud < Formula
+  desc "The official tool to managment your ucloud services"
+  homepage "https://www.ucloud.cn"
+  url "ucloud-sdk.dl.ufileos.com/ucloud-cli-0.1.1.tar.gz"
```
Comment (lembacon) [inline Formula/ucloud.rb]: Please do not omit the URL scheme.

File: Formula/ucloud.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/ucloud/ucloud-cli"
+    dir.install buildpath.children
+    cd dir do
+      system "go", "build", "-o", bin/"ucloud"
+    end
+  end
+
+  test do
+    system "#{bin}/ucloud", "version"
```
Comment (lembacon) [inline Formula/ucloud.rb]: Please try to write a test that exercises a deeper level of functionality. Thanks!

File: Formula/ucloud.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  sha256 "f41aaa6bf8063b3ccd0c5e41d3dc90f46f9fb45066513da8e306456f1f332071"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/ucloud/ucloud-cli"
+    dir.install buildpath.children
+    cd dir do
+      system "go", "build", "-o", bin/"ucloud"
```
Comment (DomT4) [inline Formula/ucloud.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

Comment (lixiaojun629) [reply Formula/ucloud.rb]: Thanks for your time!

Comment (DomT4) [reply Formula/ucloud.rb]: Thanks for the quick fix!

File: Formula/ucloud.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "#{bin}/ucloud", "config", "ls"
```
Comment (fxcoudert) [inline Formula/ucloud.rb]: Can we match (part of) the output of the test? `assert_match` or `assert_equal`

Comment (lixiaojun629) [reply Formula/ucloud.rb]: >```
>test do
>   system "#{bin}/kube-aws"
>    system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
>           "--external-dns-name", "dns", "--region", "us-west-1",
>           "--availability-zone", "zone", "--key-name", "key",
>           "--kms-key-arn", "arn", "--no-record-set",
>           "--s3-uri", "s3://examplebucket/mydir"
>    cluster_yaml = (testpath/"cluster.yaml").read
>    assert_match "clusterName: test-cluster", cluster_yaml
>    assert_match "dnsName: dns", cluster_yaml
>    assert_match "region: us-west-1", cluster_yaml
>    assert_match "availabilityZone: zone", cluster_yaml
>    assert_match "keyName: key", cluster_yaml
>    assert_match "kmsKeyArn: \"arn\"", cluster_yaml
>    installed_version = shell_output("#{bin}/kube-aws version 2>&1")
>    assert_match "kube-aws version v#{version}", installed_version
> end
```
I found an example.   I will have a try.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @lixiaojun629 for the pull request!

**Pull request closed without merging**